lib/stages/oneDiskVDIFWrite.cpp

```diff
@@ -0,0 +1,338 @@
+#include <string>      // for string
+#include <fcntl.h>     // for open, O_CREAT, O_WRONLY
+#include <unistd.h>    // for write, close, gethostname, ssize_t
+#include <visUtil.hpp> // for frameID, modulo
+
+using kotekan::bufferContainer;
+using kotekan::Config;
+using kotekan::Stage;
+using std::string;
+
+REGISTER_KOTEKAN_STAGE(oneDiskVDIFWrite);
+
+oneDiskVDIFWrite::oneDiskVDIFWrite(Config& config_, const std::string& unique_name,
+                               bufferContainer& buffer_container) :
+    Stage(config_, unique_name, buffer_container, std::bind(&oneDiskVDIFWrite::main_thread, this)) {
+    double int_year;
+    double frac_year;
+    tm ref_time{};
+
+    disk_base = config.get<std::string>(unique_name, "disk_base");
+    disk_set = config.get<std::string>(unique_name, "disk_set");
+    disk_id = config.get<int>(unique_name, "disk_id");     
+    file_name = config.get<std::string>(unique_name, "file_name");
+    file_ext = config.get_default<std::string>(unique_name, "file_ext", "vdif");
+    use_abs_path = config.get_default<bool>(unique_name, "use_abs_path", false);
+    abs_path = config.get_default<std::string>(unique_name, "abs_path", "");
+    instrument_name =
+        config.get_default<std::string>(unique_name, "instrument_name", "no_name_set");
+    write_to_disk = config.get_default<bool>(unique_name, "write_to_disk", true);
+    // TODO change to frames per frequency
+    vdif_frames_per_freq = config.get<uint32_t>(unique_name, "vdif_frames_per_freq");
+    vdif_frame_header_size = config.get<uint32_t>(unique_name, "vdif_frame_header_size");
+    num_pol = config.get<uint32_t>(unique_name, "num_pol"); 
+    exit_after_n_files = config.get_default<size_t>(unique_name, "exit_after_n_files", 0);
+    // The refernce epoch would be the last two digit or the year. If it in half of a 
+    // year, add 0.5. 
+    ref_year = config.get<double>(unique_name, "ref_year");
+    frac_year = modf(ref_year, &int_year);
+    ref_time.tm_year = (int)int_year - 1900;
+    ref_time.tm_mday = 1;
+    if (frac_year > 0){ // The reference time is in the second half of the year
+	assert(frac_year == 0.5);
+        ref_time.tm_mon = 6;
+    }
+    time_t ref_tt = std::mktime(&ref_time);
+    ref_ct = timespec{ref_tt, 0};
+    //unix_time_sec_ref = 
+    vdif_samples_per_frame = config.get_default<float>(unique_name, "vdif_samples_per_frame", 625);
+    vdif_freq_per_frame = config.get_default<float>(unique_name, "vdif_freq_per_frame", 4);
+    string note = config.get_default<std::string>(unique_name, "note", "");
+    auto& tel = Telescope::instance();
+    time_res_nsec = tel.seq_length_nsec();
+    // Set up buffers
+    in_buf = get_buffer("in_buf");
+    register_consumer(in_buf, unique_name.c_str());
+}
+
+oneDiskVDIFWrite::~oneDiskVDIFWrite(){}
+
+
+void oneDiskVDIFWrite::main_thread(){
+    int fd;
+    int payload_ctr;
+    uint8_t* input = nullptr;
+    frameID in_frame_id(in_buf);
+    uint8_t* vdif_frame;
```

Comment:
It would be good practice to set all your pointers to `nullptr` when you declare them.